src/Ssh/Ssh/Common/BastionUtils.cs

```diff
@@ -0,0 +1,318 @@
+        {
+            int port = 22;
+            string location = nic.Location;
+            CheckValidBastionDeveloperLocation(location);
+
+            string vmSubscriptionID = nic.VirtualMachine.Id;
+
+
+            string vNetId = null;
+            string vNetName = null;
+            string vNetResourceGroup = null;
+            if (nic.IpConfigurations != null && nic.IpConfigurations.Any())
+            {
+                foreach (var ipConfig in nic.IpConfigurations)
+                {
+                    if (ipConfig.Subnet != null)
+                    {
+                        string subnetId = ipConfig.Subnet.Id;
+
+                        (vNetId, vNetName, vNetResourceGroup) = GetVNetDetailsFromSubnetId(subnetId);
+                        break;
+                    }
+                }
+            }
+            ResourceGraphUtils resourceGraphUtils = new ResourceGraphUtils(context);
+            string bastionsFoundInVNet = resourceGraphUtils.QueryResourceGraph(vNetId);
+            string bastionNameInVNet = ParseAvailableBastions(bastionsFoundInVNet);
+
+            BastionHost bastion;
+
+
+            if (bastionNameInVNet == null)
+            {
+                string bastionName = vNetName + "-bastion";
+                bastion = CreateDeveloperBastion(vNetResourceGroup, bastionName, location, vNetId);
+
+            }
+            else
+            {
+                bastion = FetchDeveloperBastion(vNetResourceGroup, bastionNameInVNet);
+
+            }
+            int bastionPort = 0;  
+            string bastionEndPoint = null;
+            try
+            {
+
+
+                bastionEndPoint = GetDataPodEndPoint(bastion, context, vmSubscriptionID, port);
+
+            }
+            catch (Exception ex)
+            {
+                string error = "Error fetching end point: " + ex.Message;
+                throw new AzPSCloudException(error);
+            }
+
+            try
+            {
+                TunnelServer tunnel = new TunnelServer(context, port, bastion, bastionEndPoint, vmSubscriptionID, bastionPort);
+
+                Thread tunnelThread = new Thread(() => tunnel.StartServer());
+                tunnelThread.Start();
+
+                sshProcess.Start();
+                sshProcess.WaitForExit();
+                int sshExitCode = sshProcess.ExitCode;
+                return sshExitCode;
+            }
+            catch (Exception ex)
+            {
+                string error = "Error connecting to Bastion Host: " + ex.Message;
+                throw new AzPSCloudException(error);
+            }
+
+        }
+
+        public BastionHost FetchDeveloperBastion(string resourceGroupName, string name)
+        {
+            try
+            {
+                var bastion = this.BastionClient.Get(resourceGroupName, name);
+                if (bastion.Sku.Name != "Developer")
```

Comment:
In general, it's better to have these types of string comparisons (particularly since you also use it below) to be a const defined somewhere.